packages/yoroi-extension/app/components/wallet/create-wallet/VerifyRecoveryPhraseStep.js

```diff
@@ -0,0 +1,239 @@
+      <Stack direction="column" alignItems="left" justifyContent="center" maxWidth="690px">
+        <Typography mb="16px">
+          <FormattedHTMLMessage {...messages.description} />
+        </Typography>
+
+        <Box className={styles.verifyRecoveryPhraseArea}>
+          <Stack
+            gap="8px"
+            p="16px 14px"
+            flexDirection="row"
+            flexWrap="wrap"
+            alignItems="center"
+            justifyContent="center"
+          >
+            {enteredRecoveryPhrase.map((word, idx) => (
+              <Stack
+                item
+                // eslint-disable-next-line react/no-array-index-key
+                key={idx}
+              >
+                <Box
+                  sx={{
+                    whiteSpace: 'nowrap',
+                    overflow: 'hidden',
+                    textOverflow: 'ellipsis',
+                    display: 'flex',
+                    flexDirection: 'row',
+                    alignItems: 'center',
+                    justifyContent: 'flex-start',
+                    width: '120px',
+                    height: '40px',
+                  }}
+                  variant="body1"
+                  color="primary.200"
+                >
+                  <Typography
+                    variant="body1"
+                    color="primary.200"
+                    width="20px"
+                    sx={{
+                      display: 'flex',
+                      flexDirection: 'row',
+                      alignItems: 'center',
+                      justifyContent: 'center',
+                    }}
+                  >
+                    {idx + 1}.
+                  </Typography>
+                  {word && (
+                    <Typography
+                      sx={{
+                        background: 'linear-gradient(269.97deg, #E4E8F7 0%, #C6F7ED 99.98%)',
+                        width: '100px',
+                        height: '40px',
+                        textAlign: 'center',
+                        display: 'flex',
+                        alignItems: 'center',
+                        justifyContent: 'center',
+                        borderRadius: '8px',
+                        ml: '4px',
+                      }}
+                    >
+                      {word}
+                    </Typography>
+                  )}
+                </Box>
+              </Stack>
+            ))}
+          </Stack>
+        </Box>
+
+        <Stack
+          flexDirection="row"
+          flexWrap="wrap"
+          alignItems="center"
+          justifyContent="center"
+          gap="8px"
+        >
+          {sortedRecoveryPhrase.map(({ word, id, originalIdx }) => {
+            return (
+              <button
+                type="button"
+                key={id}
+                className={classnames(styles.wordChip, {
+                  [styles.wordAdded]: isWordAdded(word),
+                  [styles.wrongWord]: wrongWord === word,
```

Comment:
Same thing , just the string word itself is not enough, there will be duplicate words, use combined keys like `${word}-${originalIdx}` to make them unique word identifiers, or even just store the original index of the word , e.g. 

```
[styles.wrongWord]: wrongWordOriginalIdx === originalIdx,
```